modules/ppcp-settings/resources/js/data/features/actions.js

```diff
@@ -0,0 +1,135 @@
+ *
+ * @return {Action} The action.
+ */
+export const reset = () => ( { type: ACTION_TYPES.RESET } );
+
+/**
+ * Persistent. Set the full store details during app initialization.
+ *
+ * @param {{data: {}, flags?: {}}} payload
+ * @return {Action} The action.
+ */
+export const hydrate = ( payload ) => ( {
+	type: ACTION_TYPES.HYDRATE,
+	payload,
+} );
+
+/**
+ * Generic transient-data updater.
+ *
+ * @param {string} prop  Name of the property to update.
+ * @param {any}    value The new value of the property.
+ * @return {Action} The action.
+ */
+export const setTransient = ( prop, value ) => ( {
+	type: ACTION_TYPES.SET_TRANSIENT,
+	payload: { [ prop ]: value },
+} );
+
+/**
+ * Generic persistent-data updater.
+ *
+ * @param {string} prop  Name of the property to update.
+ * @param {any}    value The new value of the property.
+ * @return {Action} The action.
+ */
+export const setPersistent = ( prop, value ) => ( {
+	type: ACTION_TYPES.SET_PERSISTENT,
+	payload: { [ prop ]: value },
+} );
+
+/**
+ * Transient. Marks the store as "ready", i.e., fully initialized.
+ *
+ * @param {boolean} isReady
+ * @return {Action} The action.
+ */
+export const setIsReady = ( isReady ) => setTransient( 'isReady', isReady );
+
+/**
+ * Thunk action creator. Triggers the persistence of store data to the server.
+ *
+ * @return {Function} The thunk function.
+ */
+export function persist() {
+	return async ( { select } ) => {
+		await apiFetch( {
+			path: REST_PERSIST_PATH,
+			method: 'POST',
+			data: select.persistentData(),
+		} );
+	};
+}
+
+export function fetchFeatures( setActiveModal ) {
+	return async () => {
+		const response = await apiFetch( { path: REST_PERSIST_PATH } );
+		const features = response?.data || [];
+		const featuresWithModal = features.map( ( feature ) => ( {
+			...feature,
+			buttons: feature.buttons.map( ( button ) => ( {
+				...button,
+				onClick:
+					featureButtonOnClickMap[ button.onClick ]( setActiveModal ),
+			} ) ),
+		} ) );
+		dispatch( setFeatures( featuresWithModal ) );
+	};
+}
+
+const featureButtonOnClickMap = {
+	save_paypal_and_venmo: () => {
+		selectTab( TAB_IDS.SETTINGS, 'ppcp--save-payment-methods' );
+	},
+	advanced_credit_and_debit_cards: ( setActiveModal ) => {
+		selectTab( TAB_IDS.PAYMENT_METHODS, 'ppcp-card-payments-card' ).then(
+			() => {
+				setActiveModal( 'ppcp-credit-card-gateway' );
+			}
+		);
+	},
+	alternative_payment_methods: () => {
+		selectTab( TAB_IDS.PAYMENT_METHODS, 'ppcp-alternative-payments-card' );
+	},
+	google_pay: ( setActiveModal ) => {
+		selectTab( TAB_IDS.PAYMENT_METHODS, 'ppcp-card-payments-card' ).then(
+			() => {
+				setActiveModal( 'ppcp-googlepay' );
+			}
+		);
+	},
+	apple_pay: ( setActiveModal ) => {
+		selectTab( TAB_IDS.PAYMENT_METHODS, 'ppcp-card-payments-card' ).then(
+			() => {
+				setActiveModal( 'ppcp-applepay' );
+			}
+		);
+	},
+	pay_later: () => {
+		selectTab( TAB_IDS.PAY_LATER_MESSAGING );
+	},
+};
```

Comment:
I believe we can move this to the definition file and handle it like todos. More info in: PCP-4174